### PR TITLE
setChannelByArray proposal

### PR DESCRIPTION
**Pull request by harryprayiv**
I am sure I implemented this incredibly poorly but I think it sort of demonstrates what I would love to see from this library.  I am putting most of my number crunching power (that already drives an interface anyway) into an external machine (smart phone) that sends data to the Teensy through BLE.  So, a simple way to parse an array (with a required 512 channel input?) into straight DMX signal without having to break up my channel-setting function calls might prove useful.  

It also allows multiple channels to be changed to different levels in one fell swoop.

### CODE REVIEW

File: TeensyDmx.cpp [hunk 1]
```diff
@@ -237,6 +237,23 @@ void TeensyDmx::setChannels(
     }
 }
 
+// I am sure there is some code problems here but I wanted to suggest this feature for you to implement better than I can.
+void TeensyDmx::setChannelByArray(const uint8_t* values[]){
```
Comment (chrisstaite) [inline TeensyDmx.cpp:241]: This code will write over all sorts of memory.  I think you only want the middle loop.  The code as it stands will not work and will break the structure around it.

**Comment by harryprayiv** [comment]
I realize that this code is janky at best.  However, I submitted it as a way to describe what I want as I’m actually completely unable to test it at the moment.  I hope you understand the jist of what I was going for and can, perhaps, take it from here and make it work if you think it is a worthy addition to your library.

**Comment by chrisstaite** [comment]
I'm not particularly a fan of exposing an interface like this as it means that you are using twice the amount of RAM required.  Can you show a use-case that this is for that can't be solved by simply passing a reference to the TeensyDmx class so the get and set can be called on that?

**Comment by peternewman** [comment]
We also already have setChannels which lets you pass in a pointer and getBuffer which accesses the most up to date buffer (and the active one in the case of DMX Tx, so you could just use this to write your data direct into TeensyDMX.

**Pull request closed without merging**